$MirOS: ports/devel/autoconf/2.63/patches/patch-bin_autom4te_in,v 1.1 2009/07/15 16:43:35 bsiegert Exp $
--- bin/autom4te.in.orig	Thu Aug 28 04:08:18 2008
+++ bin/autom4te.in	Tue Nov 17 14:12:53 2009
@@ -88,15 +88,16 @@ my @include;
 my $freeze = 0;
 
 # $M4.
-my $m4 = $ENV{"M4"} || '@M4@';
-# Some non-GNU m4's don't reject the --help option, so give them /dev/null.
-fatal "need GNU m4 1.4 or later: $m4"
-  if system "$m4 --help </dev/null 2>&1 | grep reload-state >/dev/null";
+my $m4 = '@M4@';
 
+if (!grep (/gm4$/, $m4)) {
+  $m4 .= " -g -D__gnu__";
+} else {
 # Set some high recursion limit as the default limit, 250, has already
 # been hit with AC_OUTPUT.  Don't override the user's choice.
 $m4 .= ' --nesting-limit=1024'
   if " $m4 " !~ / (--nesting-limit(=[0-9]+)?|-L[0-9]*) /;
+}
 
 
 # @M4_BUILTIN -- M4 builtins and a useful comment.
@@ -462,11 +463,10 @@ sub handle_m4 ($@)
   # We don't output directly to the cache files, to avoid problems
   # when we are interrupted (that leaves corrupted files).
   xsystem ("$m4"
-	   . join (' --include=', '', map { shell_quote ($_) } @include)
-	   . ' --debug=aflq'
-	   . (!exists $ENV{'AUTOM4TE_NO_FATAL'} ? ' --fatal-warning' : '')
-	   . " @M4_DEBUGFILE@=" . shell_quote ("$tcache" . $req->id . "t")
-	   . join (' --trace=',   '', sort @macro)
+	   . join (' -I ', '', map { shell_quote ($_) } @include)
+	   . ' -daflq'
+ 	   . " -o$tcache" . $req->id . "t"
+	   . join (' -t',   '', sort @macro)
 	   . " " . files_to_options (@ARGV)
 	   . " > " . shell_quote ("$ocache" . $req->id . "t"));
 
@@ -930,8 +930,7 @@ sub freeze ($)
   # When processing the file with diversion disabled, there must be no
   # output but comments and empty lines.
   my $result = xqx ("$m4"
-		    . ' --fatal-warning'
-		    . join (' --include=', '', map { shell_quote ($_) } @include)
+		    . join (' -I ', '', map { shell_quote ($_) } @include)
 		    . ' --define=divert'
 		    . " " . files_to_options (@ARGV)
 		    . ' </dev/null');
@@ -944,8 +943,7 @@ sub freeze ($)
   # If freezing produces output, something went wrong: a bad `divert',
   # or an improper paren etc.
   xsystem ("$m4"
-	   . ' --fatal-warning'
-	   . join (' --include=', '', map { shell_quote ($_) } @include)
+	   . join (' -I ', '', map { shell_quote ($_) } @include)
 	   . " --freeze-state=" . shell_quote ($output)
 	   . " " . files_to_options (@ARGV)
 	   . ' </dev/null');
